### DISCUSSION TITLE
onPublish & onUnPublish hooks

### DISCUSSION THREAD

**Discussion by johannesschaffer** [payloadcms/payload#1853] [Feature Requests & Ideas]
Currently, collection hooks don't include onPublish & onUnPublish. Those are however required when used as webhooks to rebuild a SSG rendered app.

**Comment by DanRibbens** [comment]
I don't think you would need a specific `onPublish`. You can use existing hooks (`afterchange`, `beforechange`, `afterRead`, etc) and add your conditional logic as needed using the `doc.status` and `previousDoc.status`.

https://payloadcms.com/docs/hooks/collections#afterchange
We should consider adding more documentation to cover this scenario, as I imagine it would be fairly common.

What are your thoughts on this?

**Reply by johannesschaffer** [reply]
Yeah, makes sense;) Didn't think of that. 
Two things:
1. It's _status, not status - just for reference, in case someone finds this thread here
2. Maybe add typings to "doc" for the default fields like "_status", so it's easier discoverable with TS? Currently it autocompletes doc.doc.doc.... That's a bug I guess?

**Reply by tlmader** [reply]
@DanRibbens Editing a published doc and triggering an autosave also results in `previousDoc._status: 'published'` and `doc._status: 'draft'`. Is there a recommended approach in 2025 for handling unpublish actions? Our use case is removing a document from our search index when it is unpublished.
